src/clock/mumeclock.cpp

```diff
@@ -199,92 +149,100 @@ void MumeClock::parseMumeTime(const QString &mumeTime, const int64_t secsSinceEp
         log("Detected new Mume start epoch " + QString::number(newStartEpoch) + " ("
             + QString::number(newStartEpoch - m_mumeStartEpoch) + " seconds from previous)");
     } else {
-        log("Synchronized clock using 'time' output.");
+        log("Synchronized clock using 'time' output");
     }
     if (weekDay != capturedMoment.weekDay()) {
         qWarning() << "Calculated week day does not match MUME";
     }
     m_mumeStartEpoch = newStartEpoch;
 }
 
-void MumeClock::parseWeather(const QString &str)
+void MumeClock::slot_parseGmcpInput(const GmcpMessage &msg)
 {
-    const int64_t secsSinceEpoch = QDateTime::QDateTime::currentDateTimeUtc().toSecsSinceEpoch();
-    parseWeather(str, secsSinceEpoch);
-}
+    if (!(msg.isEventMoon() || msg.isEventDarkness() || msg.isEventSun())
+        || !msg.getJsonDocument().has_value() || !msg.getJsonDocument()->isObject())
+        return;
 
-void MumeClock::parseWeather(const QString &str, int64_t secsSinceEpoch)
-{
-    if (!m_stringTimeHash.contains(str)) {
+    const QJsonObject obj = msg.getJsonDocument()->object();
+    if (!obj.contains("what") || !obj.value("what").isString())
         return;
+
+    const auto type = obj["what"].toString();
+    if (type.isEmpty())
+        return;
+
+    MumeTimeEnum time = MumeTimeEnum::UNKNOWN;
+    if (msg.isEventSun()) {
+        switch (type.at(0).toLatin1()) {
+        case 'l': // light
+            time = MumeTimeEnum::DAY;
+            break;
+        case 'd': // dark
+            time = MumeTimeEnum::NIGHT;
+            break;
+        case 'r': // rise
+            time = MumeTimeEnum::DAWN;
+            break;
+        case 's': // set
+            time = MumeTimeEnum::DUSK;
+            break;
+        default:
+            assert(false);
```

Comment:
It will throw an assertion (aka crash) during a debug build only. We want this by design, rather than it silently failing. I suppose there is no harm in adding a warning but should be pretty clear from the core dump or assertion point.